src/rust/src/orchard_keys_ffi.rs

```diff
@@ -0,0 +1,304 @@
+use std::io::{Read, Write};
+use std::slice;
+use tracing::error;
+
+use orchard::keys::{DiversifierIndex, FullViewingKey, IncomingViewingKey, SpendingKey};
+use orchard::Address;
+
+use crate::streams_ffi::{CppStreamReader, CppStreamWriter, ReadCb, StreamObj, WriteCb};
+
+//
+// Addresses
+//
+
+#[no_mangle]
+pub extern "C" fn orchard_address_clone(addr: *const Address) -> *mut Address {
+    unsafe { addr.as_ref() }
+        .map(|addr| Box::into_raw(Box::new(*addr)))
+        .unwrap_or(std::ptr::null_mut())
+}
+
+#[no_mangle]
+pub extern "C" fn orchard_address_free(addr: *mut Address) {
+    if !addr.is_null() {
+        drop(unsafe { Box::from_raw(addr) });
```

Comment:
The explicit call to `drop` is unnecessary; see the examples in the `Box::from_raw` [documentation](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.from_raw).
```suggestion
        unsafe { Box::from_raw(addr) };
```

We do call `drop` explicitly (and also I believe unnecessarily) in some of our other FFI code, but not [here](https://github.com/zcash/zcash/blob/91bb1d3b46e44338ebd06ce04e1f38e756e85f82/src/rust/src/address_ffi.rs#L225-L228) for example.

Reply:
I think that the `let _ =` idiom is perhaps less clear with respect to the intent of the operation, whereas `drop` is completely explicit, so I prefer to keep this as it is (and perhaps change the other locations where we're explicitly freeing memory to use an explicit drop as well.)